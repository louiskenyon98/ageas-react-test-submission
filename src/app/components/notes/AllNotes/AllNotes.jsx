import React from 'react';
import PropTypes from 'prop-types';
import Card from '../Card/Card';

class AllNotes extends React.PureComponent {
    renderAll() {
        return this.props.data.map((note) => {
            return (
                <Card
                    key={note.id}
                    id={note.id}
                    title={note.title}
                    body={note.body}
                    delete={this.props.confirmDelete}
                />
            )
        })
    }

    render() {
        return (
            <div className={`container-fluid`}>
                <div className={`row`}>
                    {this.renderAll()}
                </div>
            </div>
        )
    }
}

AllNotes.propTypes = {
    data: PropTypes.array
};

export default AllNotes;
