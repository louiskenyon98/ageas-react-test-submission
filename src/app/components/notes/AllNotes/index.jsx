import React from 'react';
import {Link} from 'react-router-dom';
import styles from '../../../../themes/style.scss';

class AllNotes extends React.Component {
    renderAll() {
        //Map array in reverse order so as to render the newest notes first.
        //Shallow copy so as not to alter original array.
        return this.props.notes.slice(0).reverse().map((note) => {
            return (
                <div className={`col-xs-1 col-md-4 col-xl-3 d-flex align-items-stretch ${styles.column}`} key={note.id}>
                    <div className={`ui raised card ${styles.card}`}>
                        <div className="content">
                            <i
                                onClick={() => this.props.deleteNote(note.id)}
                                className={`right floated close icon ${styles["delete-button"]}`}
                            />
                            <div className={`header ${styles.header}`}>
                                {note.title}
                            </div>
                            <div className={`description ${styles.body}`}>
                                <p>{note.body}</p>
                            </div>
                        </div>
                        <div className="extra content">
                            <span className="right floated pencil alternate">
                                <Link to={`/edit/${note.id}`}>
                                    <i className={`pencil alternate icon ${styles["edit-button"]}`}/>
                                </Link>
                            </span>
                        </div>
                    </div>
                </div>
            )
        })
    }

    render() {
        return (
            <div className={`container-fluid`}>
                <div className={`row`}>
                    {this.renderAll()}
                </div>
            </div>
        )
    }
}

export default AllNotes;
