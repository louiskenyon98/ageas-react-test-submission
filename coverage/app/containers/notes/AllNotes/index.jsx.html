<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/containers/notes/AllNotes/index.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="..\..\..\..\prettify.css" />
    <link rel="stylesheet" href="..\..\..\..\base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(..\..\..\..\sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="..\..\..\..\index.html">All files</a> / <a href="index.html">app/containers/notes/AllNotes</a> index.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/4</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import {connect} from 'react-redux';
&nbsp;
import {getAllNotesSelector} from '../../../selectors/note/note.selector';
import {getNotes, deleteNote} from '../../../actions/note/notes.action';
import {showDeleteNoteConfirmationModal} from '../../../actions/common/modal.action';
&nbsp;
import AllNotes from '../../../components/notes/AllNotes';
&nbsp;
export class AllNotesContainer extends React.Component {
&nbsp;
    componentDidMount() {
        this.props.getNotes();
    }
&nbsp;
    render() {
        return (
            &lt;AllNotes
                data={this.props.notes}
                confirmDelete={this.props.showDeleteNoteConfirmationModal}
            /&gt;
        )
    }
&nbsp;
}
&nbsp;
const mapStateToProps = (state) =&gt; {
    return getAllNotesSelector(state)
};
&nbsp;
export default connect(mapStateToProps, {getNotes, deleteNote, showDeleteNoteConfirmationModal})(AllNotesContainer);
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Apr 23 2019 10:38:36 GMT+0100 (GMT Daylight Time)
</div>
</div>
<script src="..\..\..\..\prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="..\..\..\..\sorter.js"></script>
</body>
</html>
