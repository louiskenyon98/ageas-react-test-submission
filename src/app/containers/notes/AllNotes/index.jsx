import React from 'react';
import {connect} from 'react-redux';
import {getNotes, deleteNote} from '../../../actions';

import AllNotes from "../../../components/notes/AllNotes";
// import AllNotes from "../../components/AllNotes";

class AllNotesContainer extends React.Component {
    componentDidMount() {
        this.props.getNotes();
    }
    render() {
        return (
            <AllNotes
                notes={this.props.notes}
                deleteNote={this.props.deleteNote}
            />
        )
    }

}

const mapStateToProps = (state) => {
    return {
        notes: Object.values(state.notes)
    }
};

export default connect(mapStateToProps, {getNotes, deleteNote})(AllNotesContainer);
