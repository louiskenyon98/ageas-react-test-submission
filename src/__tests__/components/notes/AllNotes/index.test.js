import React from 'react';
import {shallow} from 'enzyme';
import AllNotes from '../../../../app/components/notes/AllNotes/index';

describe('AllNotes', () => {
    let props = {};
    let wrapper = () => shallow(<AllNotes {...props}/>);
    beforeEach(() => {
        props = {
            data: [
                {
                    id: 1,
                    title: 'noteTitle',
                    body: 'noteBody',
                },
                {
                    id: 2,
                    title: 'noteTitle2',
                    body: 'noteBody2',
                },
                {
                    id: 3,
                    title: 'noteTitle3',
                    body: 'noteBody3',
                }],
            confirmDelete: jest.fn()
        }
    });
    describe('render', () => {
        it('should render a card component with props', () => {
            expect(wrapper()).toMatchSnapshot();
        });
    });
    describe('functionality', () => {
        it('should not call confirmDelete', () => {
            expect(props.confirmDelete).not.toHaveBeenCalled();
            wrapper().find('Card').first().props().delete();
            expect(props.confirmDelete).toHaveBeenCalled();
        })
    })
});
//:todo tests
