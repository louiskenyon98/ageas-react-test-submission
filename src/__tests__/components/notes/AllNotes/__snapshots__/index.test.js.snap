// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllNotes render should render a card component with props 1`] = `
<div
  className="container-fluid"
>
  <div
    className="row"
  >
    <Card
      body="noteBody"
      delete={[MockFunction]}
      id={1}
      key="1"
      title="noteTitle"
    />
    <Card
      body="noteBody2"
      delete={[MockFunction]}
      id={2}
      key="2"
      title="noteTitle2"
    />
    <Card
      body="noteBody3"
      delete={[MockFunction]}
      id={3}
      key="3"
      title="noteTitle3"
    />
  </div>
</div>
`;
